File: ui/react/src/action/AddChart.ts
import Chart from '../model/Chart';
import SheetAction from './SheetAction';
import Action from './Action';

class AddChart extends SheetAction {
    public chart: Chart;

    public static deserialize(input: any) {
        const sheetName: string = input.sheetName;
        const chart: Chart = Chart.deserialize(input.chart);
        return new AddChart(sheetName, chart);
    }

    constructor(sheetName: string, chart: Chart) {
        super(sheetName);
        this.chart = chart;
    }

    public wrapper(): Action {
        const action = new Action();
        action.addChart = this;
        return action;
    }

}

export default AddChart;

File: core/src/main/java/com/ctrip/ferriswheel/core/action/AddChart.java
package com.ctrip.ferriswheel.core.action;

import com.ctrip.ferriswheel.common.action.Action;
import com.ctrip.ferriswheel.common.chart.Chart;

public final class AddChart extends SheetAction implements Action {
    private Chart chartData;

    public AddChart() {
    }

    public AddChart(String sheetName, Chart chartData) {
        super(sheetName);
        this.chartData = chartData;
    }

    @Override
    public String toString() {
        return "AddChart{" +
                "sheetName='" + getSheetName() + '\'' +
                ", chartData=" + chartData +
                '}';
    }

    public Chart getChartData() {
        return chartData;
    }

    public void setChartData(Chart chartData) {
        this.chartData = chartData;
    }
}
